### ISSUE TITLE
TypeError: Can't convert ObjectHelper (or subclass) when scaling a page

### ISSUE THREAD

**Issue by grzebo** [pdfarranger/pdfarranger#599]
Describe the bug
----------------
I'm using latest Kubuntu (21.10). Bundled version of PDF Arranger is 1.6.2-3.
When I try to save a PDF file after it's been modified (any kind of modification: rotating a page, deleting a page, adding a page etc.), I get an error: TypeError: Can't convert ObjectHelper (or subclass) to Object implicitly. Use .obj to get access the underlying object.

Bug doesn't appear when I downgrade to pdfarranger_1.4.2-1_all.deb from Ubuntu LTS.

To Reproduce
------------

1.  Open a PDF file in PDF Arranger
2.  Modify it in any way (i.e. delete a page)
3.  Click Save or Save As...
4.  Error dialog appears: TypeError: Can't convert ObjectHelper (or subclass) to Object implicitly. Use .obj to get access the underlying object.
5. The file doesn't get saved.

Expected behavior
-----------------
File should get saved.

Input files
-----------
Error occurs when saving any kind of file.

Screenshots
-----------
$ pdfarranger 
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/pdfarranger/pdfarranger.py", line 976, in choose_export_pdf_name
    self.save(mode, file_out)
  File "/usr/lib/python3/dist-packages/pdfarranger/pdfarranger.py", line 160, in wrapper
    func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/pdfarranger/pdfarranger.py", line 1050, in save
    exporter.export(self.pdfqueue, to_export, file_out, mode, m)
  File "/usr/lib/python3/dist-packages/pdfarranger/exporter.py", line 165, in export
    new_page = pdf_output.make_indirect(new_page)
TypeError: Can't convert ObjectHelper (or subclass) to Object implicitly. Use .obj to get access the underlying object.

System and Versions
-------------------

-   PDF Arranger version and where you get it: 1.6.2-3, official Kubuntu repo, bug also appears on version 1.8.0-1 from Debian repos
-   pikepdf version and where you get it: python3-pikepdf in version 1.17.3+dfsg-5ubuntu1 (official Kubuntu repo)
-   OS name and version: Kubuntu 21.10

**Comment by jeromerobert** [comment]
Could you provide the versions of pikepdf and libqpdf advertised in the *About* menu of PDF Arranger ?

**Comment by grzebo** [comment]
(PDF Arranger uses libqpdf 10.4.0 and pikepdf 4.2.0)

**Comment by jeromerobert** [comment]
> pikepdf version and where you get it: python3-pikepdf in version 1.17.3+dfsg-5ubuntu1 (official Kubuntu repo)

So this is wrong. May be you have this package installed but PDF Arranger is not using it. You probably did some installation with `pip` and have versions of qpdf, pikepdf and PDF Arranger in `$HOME/.local`. Maybe you should clean this up first.

**Comment by jeromerobert** [comment]
I'm able to reproduce the bug. With pikepdf 4.2.0 installed with `pip` and I also need to scale a page (other changes does not trigger the bug).

**Comment by grzebo** [comment]
This is it! manually uninstalling pikepdf using pip and reinstalling the distro version of python3-pikepdf fixed the problem.

```
# pip uninstall pikepdf
# aptitude reinstall python3-pikepdf
```
I still don't know how I got pikepdf 4.2.0 installed - I am not a python user, I don't use pip nor do I install python packages. I think that it was somehow upgraded when I upgraded from Kubuntu 21.04 to 21.10 and that broke my pdfarranger setup.

**Comment by jeromerobert** [comment]
I also tested pikepdf 1.17.3 and qpdf 10.1.0 and it works. A fix for pikepdf 4.2 is:
```patch
--- a/pdfarranger/exporter.py
+++ b/pdfarranger/exporter.py
@@ -91,7 +91,7 @@ def _scale(doc, page, factor):
         # unset it on the input page before
         rotate = page.Rotate
         page.Rotate = 0
-    page = doc.make_indirect(page)
+    page = doc.make_indirect(page.obj)
     page_id = len(doc.pages)
     newmediabox = [factor * float(x) for x in page.MediaBox]
     content = "q {} 0 0 {} 0 0 cm /p{} Do Q".format(factor, factor, page_id)
```

Now the most tedious part will be to find a solution that works for all versions of pikepdf between 1.17 and 4.2
